### ISSUE TITLE
Bug on 57a72fd with field_basic test case.

### ISSUE THREAD

**Issue by Deophius** [Sudoku-dev/Sudoku#74]
## Description
The test case fails, even if I reduced the data test case to just one.
Error message is like this:

```
Running 114 test cases...
/home/Deophius/Sudoku_proj/Sudoku/History/Test/field_conv.inl(28): error: in "history_file/field_basic/_0": jsonv['size'] is 9
got is 0


Failure occurred in a following context:
    sample = 10; size = 9; 
/home/Deophius/Sudoku_proj/Sudoku/History/Test/field_conv.inl(32): error: in "history_file/field_basic/_0": read.size() is0
Failure occurred in a following context:
    sample = 10; size = 9; 
/home/Deophius/Sudoku_proj/Sudoku/History/Test/field_conv.inl(33): error: in "history_file/field_basic/_0": field.size() is9
Failure occurred in a following context:
    sample = 10; size = 9; 
/home/Deophius/Sudoku_proj/Sudoku/History/Test/field_conv.inl(35): fatal error: in "history_file/field_basic/_0": critical check field.size() == read.size() has failed
Failure occurred in a following context:
    sample = 10; size = 9; 

*** 4 failures are detected in the test module "Sudoku"
```

I think this is rather shocking.

## Solve ways
1. Look at this line.
``` 
got is 0
```
This proves that nothing is read from the x and y arrays. So `xend == itx` or `yend == ity`. Why? Perhaps we need to look up the json docs.

2. Debug `Rule::Field` to see if we can find anything interesting.

**Comment by Deophius** [comment]
How stupid! Look at this:
```diff
diff --git a/History/userconv.cpp b/History/userconv.cpp
index 7d6b79a..006bc70 100644
--- a/History/userconv.cpp
+++ b/History/userconv.cpp
@@ -123,7 +123,7 @@ namespace History::File
         field.clear();
         // Work begins
         std::size_t got = 0; // Counter for how many blocks got.
-        auto itx = jsonv["x"].cbegin(), ity = jsonv["y"].cbegin();
+        auto itx = jsonv["x"].cbegin(), ity = jsonv["y"].cend();
         const auto xend = jsonv["x"].cend(), yend = jsonv["y"].cend();
         while (itx != xend && ity != yend)
         {
```

Yes, it's truly a typo.

**Comment by Deophius** [comment]
Closed in commit 8e0b552
